cppapi/server/zmqeventsupplier.cpp

```diff
@@ -1531,7 +1560,7 @@ void ZmqEventSupplier::push_event(DeviceImpl *device_impl,std::string event_type
 		{
 			if (require_wait == true)
 			{
-				push_cond.wait();
+				unlock_future.wait();
```

Comment:
After wait() returns, you should call reset(), otherwise future stays 'ready' and subsequent waits() will return immediately. 

I do not understand how current version of the patch fixes the issue ...

Reply:
Yes. It should be right after push_mutex.lock(). It seems, I missed this line while merging from local repository.
I will recheck all code.